File: Leetcode/2918 - Minimum Equal Sum of Two Arrays After Replacing Zeros/solution.js
var minSum = function (nums1, nums2) {
    let sumNums1 = 0;
    let sumNums2 = 0;
    let zeroInNums1 = 0;
    let zeroInNums2 = 0;

    for (const num of nums1) {
        if (num === 0) {
            sumNums1 += 1;
            zeroInNums1++;
        } else {
            sumNums1 += num;
        }
    }
    for (const num of nums2) {
        if (num === 0) {
            sumNums2 += 1;
            zeroInNums2++;
        } else {
            sumNums2 += num;
        }
    }

    if ((sumNums1 < sumNums2 && zeroInNums1 === 0) || (sumNums2 < sumNums1 && zeroInNums2 === 0)) {
        return -1;
    } else {
        return Math.max(sumNums1, sumNums2);
    }
};

File: Leetcode/2918 - Minimum Equal Sum of Two Arrays After Replacing Zeros/solution.py
class Solution:
    def minSum(self, nums1: List[int], nums2: List[int]) -> int:
        sumNums1, sumNums2, zeroInNums1, zeroInNums2 = (0, 0, 0, 0)

        for num in nums1:
            if (num == 0):
                sumNums1 += 1
                zeroInNums1 += 1
            else:
                sumNums1 += num

        for num in nums2:
            if (num == 0):
                sumNums2 += 1
                zeroInNums2 += 1
            else:
                sumNums2 += num

        if ((sumNums1 < sumNums2 and zeroInNums1 == 0) or (sumNums2 < sumNums1 and zeroInNums2 == 0)):
            return -1
        else:
            return max(sumNums1, sumNums2)